lua/nvim-treesitter/query_predicates.lua

```diff
@@ -71,4 +72,30 @@ M['has_ancestor?'] = function(query, match, pred)
   return false
 end
 
+M['adjacent?'] = function(query, match, pred)
+  if #pred < 3 then error("adjacent? must have at least two arguments!") end
+  local node = get_node(query, match, pred[2])
+  if not node then return true end
+
+  local adjacent_types = {unpack(pred, 3)}
+  local adjacent_node = ts_utils.get_next_node(node, false, false)
+
+  if not adjacent_node then return false end
+
+  for _, adjacent_type in ipairs(adjacent_types) do
+    print(vim.inspect(adjacent_type))
+    if type(adjacent_type) == "number" then
+      if get_node(query, match, adjacent_type) == adjacent_node then
```

Comment:
This is nice to read but implies a full loop to all children on each call. Should be tolerable.

Reply:
Are you referring to the loop of adjacent_types?

Reply:
get_node loops over all children each call. would be enough to just check the next node.

Reply:
sounds good. I'll make that change

Reply:
This get_node call just seems to do this

```lua
local function get_node(query, match, pred_item)
  return utils.get_at_path(match, query.captures[pred_item]..'.node')
end
```

Are you thinking about a different call?